src/main/java/uk/ac/ebi/atlas/hcalandingpage/HcaHumanExperimentService.java

```diff
@@ -18,11 +20,16 @@ public HcaHumanExperimentService(ExperimentTrader experimentTrader,
         this.hcaHumanExperimentDao = hcaHumanExperimentDao;
     }
 
-    public ImmutableSet<Experiment> getPublicHumanExperiments(String characteristicName, String characteristicValue) {
-        return hcaHumanExperimentDao.fetchExperimentAccessions(characteristicName, characteristicValue)
+    public ImmutableSet<Experiment> getPublicHumanExperiments(String characteristicName,
+                                                              Set<String> characteristicValues) {
```

Comment:
Same suggestion as above. Make more use of `ImmutableSet<String>` instead of `Set<String>`